Shifty.App/Components/Voucher.razor

```diff
@@ -0,0 +1,138 @@
+                </MudContainer>
+            }
+            @if (_showResult)
+            {
+                <MudTextField Text="@_voucherCodes"  
+                            @ref="_multilineReference"
+                            T="string"
+                            Adornment="Adornment.End" 
+                            Style="border-width: 2px; padding: 4px;" 
+                            Outlined="true" 
+                            AdornmentIcon="@Icons.Material.Outlined.ContentCopy"
+                            OnAdornmentClick="@(async () => await CopyToClipboard())" 
+                            Lines="@_vouchers.Count" 
+                            ReadOnly=true />
+            }
+        </MudCardContent>
+    </MudCard>
+</MudContainer>
+
+@code
+{
+    private VoucherForm _voucherForm = new();
+    private bool _isFormValid = false;
+    private bool _productsLoaded = false;
+    private bool _showResult = false;
+    private bool _showProgressBar = false;
+    private ICollection<ProductDto> _products = new List<ProductDto>();
+    private ICollection<IssueVoucherResponse> _vouchers;
+    private string _voucherCodes;
+    private MudTextField<string> _multilineReference;
+    
+    private class VoucherForm
+    {
+        [Required]
+        public string Description { get; set; }
+        public ProductDto Product { get; set; }
+        public int Amount { get; set; } = 1;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        Snackbar.Configuration.PositionClass = Defaults.Classes.Position.BottomRight;
+        (_productsLoaded, _products) = await _productService.GetProducts();
+
+        if (!_productsLoaded)
+        {
+            Snackbar.Add("Fatal error loading products", Severity.Error);
+            Console.WriteLine("Error happened while loading products...");
+        }
+    }
+
+    private async Task CopyToClipboard()
+    {
+        await JSRuntime.InvokeVoidAsync("navigator.clipboard.writeText", _voucherCodes);
+        Snackbar.Add("Codes copied to clipboard", Severity.Success);
+    }
+
+    private async Task IssueVoucher()
+    {
+        _showResult = false;
+        _showProgressBar = true;
+        (var success, _vouchers) = await _voucherService.IssueVouchers(amount: _voucherForm.Amount, productId: _voucherForm.Product.Id, description: _voucherForm.Description);
+        if (success)
+        {
+            _showProgressBar = false;
+            _voucherCodes = string.Join("\n", _vouchers.Select(response => response.VoucherCode));
+            _showResult = true;
+        }
+        else
+        {
+            _showProgressBar = false;
+            _showResult = false;
+            Snackbar.Add("Fatal error issuing voucher(s).", Severity.Error);
+        }
+    }
+    
+    Func<ProductDto,string> _converter = p => p != null ? $"{p.Name} ({p.NumberOfTickets})" : "";
```

Comment:
Is it intuitive to the user that the number represent the amount of tickets the product gives?

Reply:
Hmm, what else would you suggest?

We could do `$"{p.NumberOfTickets}x {p.Name}"` instead?

I mean, after all, it will likely be a list similar to:

- Filter Coffee (1)
- Filter Coffee (10)
- Fancy Coffee (1)
- Fancy Coffee (10)

I would argue that "in the context", it is user friendly as long as the name + amount is present 😄 Wether one or the other is more, I will let you decide so.